### DISCUSSION TITLE
SomfyPlus hardware details

### DISCUSSION THREAD

**Discussion by ViViDboarder** [HomeSpan/HomeSpan#377] [Q&A] [answered]
Just came across this project and it looks awesome! Cheaper than the Tahoma and works offline, so right up my alley.

I'm new to Arduinos but not new to software and a little bit of hardware hacking generally, but it's been a while since I've build electronics. Is there a particular resistance I should be getting for the current limiting resistors? Or a particular gauge wire I should grab?

Thanks!

**Comment by HomeSpan** [comment, marked as answer]
When connecting LEDs to 3.3V pins, I generally use 330 Ohm resistors.  This yields 10mA, which works well for most LEDs.  24 gauge wire is more than sufficient.

Note the SomfyPlus project is fairly advanced.  If you are new to the ESP32 and the Arduino IDE I suggest first starting with just  getting all the software running and ensuring you can get a hello-world sketch running correctly.  Next you can install the HomeSpan library and try out Tutorial Example 1-SimpleLightBulb.  This software-only example is critical since it tests whether you can properly connect to your network, pair the device to HomeKit, and control HomeSpan from your iPhone Home App.  Once satisfied everything is working, I'd recommend moving on to Example 5-WorkingLED, in which you'll connect an LED to the ESP and control it directly from the Home App.  If that works, you're all set for implementing SomfyPlus.

**Reply by ViViDboarder** [reply]
Thanks! That’s good advice, especially as each phase can be done with the same set of parts incrementally.
